### ISSUE TITLE
HA integration: only play/pause works

### ISSUE THREAD

**Issue by JorgeAnzola** [postlund/pyatv#401]
Hi, man! First of all, thanks for working on this! I'm a complete rookie in HA, and my Apple TV has been the first guinea pig. I managed to install HACS and the repo and to get the credentials with `atvremote` (Not the dmap creds, tho).

In the card, I see what's playing and the art cover, but only the play/pause button works, nothing else. This is my config, am I missing something?

```
apple_tv:
  - host: 192.168.178.30
    identifier: CC:**:**:**:**:**
    name: Living Room
    protocol: MRP
    credentials:
      mrp: a9d7****
      airplay: 7277****
```

If you could drop me a hint would be highly appreciated 🙌🏻

**Comment by postlund** [comment]
Hey and thanks! 🖖

Seems like most important features are working 😉 The component doesn't support many "fun" features yet. Which functionality are you testing and what are you watching/which app are you using? Is the same thing possible with the Remote app? Also, do you see any errors in the log? It's hard to debug basing it on "some features doesn't work" 🧐

**Comment by JorgeAnzola** [comment]
Everything I referred was about the card in my homepage, that the next and prev buttons don't  do anything 😰I just enabled the logs like:

```
logger:
  logs:
    pyatv: debug
```

Once I figure out how to see them, I'll share them with you.

**Comment by postlund** [comment]
What did you play (kind of media and app) at the time?

**Comment by postlund** [comment]
I tried previous/next in the Music app earlier today and that worked as expected, at least for me.

**Comment by JorgeAnzola** [comment]
Indeed with the Music app next and prev worked until it didn't, then I received this error:

```
'service': 'media_stop', 'service_data': {'entity_id': 'media_player.living_room'}, 'id': 42}
2020-01-31 09:54:17 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=media_player, service=media_stop, service_data=entity_id=media_player.living_room>
2020-01-31 09:54:17 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.xxxxxxxxxx] object NoneType can't be used in 'await' expression
```

With Spotify only the play/pause button works.

And the turn off/on never worked for me. I'll email you a more detailed log file!

**Comment by postlund** [comment]
> Indeed with the Music app next and prev worked until it didn't, then I received this error:
> 
> ```
> 'service': 'media_stop', 'service_data': {'entity_id': 'media_player.living_room'}, 'id': 42}
> 2020-01-31 09:54:17 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=media_player, service=media_stop, service_data=entity_id=media_player.living_room>
> 2020-01-31 09:54:17 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.xxxxxxxxxx] object NoneType can't be used in 'await' expression
> ```

Right, I will look in the log to see what happens.

> With Spotify only the play/pause button works.

Did you use the Apple TV app or stream via AirPlay?

> And the turn off/on never worked for me. I'll email you a more detailed log file!

On/off only turns on or off the media player in Home Assistant. Properly turning on or off the device is something I haven't been able to solve yet, so that is expected behavior.

**Comment by postlund** [comment]
From the logs, I can see this:
```

setStateMessage] {
  supportedCommands {
    supportedCommands {
      command: TogglePlayPause
      enabled: true
    }
    supportedCommands {
      command: Play
      enabled: true
    }
    supportedCommands {
      command: Pause
      enabled: true
    }
  }
  playbackQueue {
    location: 0
    ...
  }
  displayName: "Spotify"
  playbackState: Playing
  playbackQueueCapabilities {
  }
  playerPath {
    origin {
      type: Local
      displayName: "Living Room"
      identifier: 1280262988
    }
    client {
      processIdentifier: 389
      bundleIdentifier: "com.spotify.client"
    }
    player {
      identifier: "MediaRemote-DefaultPlayer"
      displayName: "Default Player"
    }
  }
  playbackStateTimestamp: 602153752.728885
}
```
Conclusion is that Spotify only supports play, pause and toggle between the two. Unfortunately nothing I can do about it ☹️

Regarding shuffle, I have honestly not tried that. How did do it? Seems like the shuffle key is not provided?

```
 2020-01-31 10:34:18 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.28861
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in han
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1204, in async_call
    processed_data = handler.schema(service_data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 208, in __call__
    return self._exec((Schema(val) for val in self.validators), v)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 287, in _exec
    raise e if self.msg is None else AllInvalid(self.msg, path=path)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 283, in _exec
    v = func(v)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 817, in validate_
    return schema(data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 594, in validate_
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 432, in validate_
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: required key not provided @ data['shuffle']
2020-01-31 10:34:53 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.28861
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 134, in han
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1204, in async_call
    processed_data = handler.schema(service_data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 208, in __call__
    return self._exec((Schema(val) for val in self.validators), v)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 287, in _exec
    raise e if self.msg is None else AllInvalid(self.msg, path=path)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/validators.py", line 283, in _exec
    v = func(v)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 817, in validate_
    return schema(data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 594, in validate_
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 432, in validate_
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: required key not provided @ data['sound_mode']
```
Also `sound_mode` is something else you call I guess? It's not something that is implemented by me at least.

**Comment by postlund** [comment]
Never mind, I haven't implemented shuffle and repeat in the component as it seems... I will have to fix that.

**Comment by postlund** [comment]
I think I have some regressions regarding shuffle and repeat. There's an enabled property that I have to look at and probaly use to disable repeat. Will investigate that once I get some time.

**Comment by romquenin** [comment]
Thanks for your great work @postlund, i hope you will find a way to turn the atv on and off.

**Comment by postlund** [comment]
@romquenin A power interface was added by @stickpin. So some sort of power support will come to the component. Can be tested with `atvremote` (`turn_on` and `turn_off`).

I have added a "features" interface as well, so it will be possible to only show controls that are available and hide everything else. Improvements will be made in near future. Use `atvremote ... features` to try it out.

Will close this issue know as I don't think we get any further.
